### PR TITLE
AIX: Implement shared_target = "aix-solib" support

### LINKED ISSUE #21518: add support for AIX shared libraries using only .so files, not using .so files inside archive files (.a)
AIX builds of OpenSSL package shared libraries as objects inside archive files, often referred to as `libssl.a(libssl.so)`
This doesn't work correctly when building with variant names.  The packaging doesn't create all the file system objects needed at runtime using the variant names.  The archive file (required at runtime) retains the non-variant name, and the object inside has the variant name.

AIX supports use of shared library files directly, not packaged inside an archive file.  If we add build support for using direct shared libraries on AIX, it will also fix the problem with variant names, since all files needed at runtime will use the variant names.

More details at discussion #21504

### PR DESCRIPTION
**Pull request by johnkohl-hcl**
This builds shared libraries as libxxx.so, libxxx.so.ver and static libraries as libxxx.a.  For shlib_variant builds, it builds libxxx.so, libxxxvariant.so.ver, and libxxxx.a.  libxxx.so is a linker import library that when causes the linker to embed a run-time dependency reference to libxxxvariant.so.ver.  Only libxxxvariant.so.ver is needed at runtime.

Fixes #21518

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
We will need a CLA for this. @johnkohl-hcl could you please submit a CLA - I assume we will need both ICLA and CCLA for you. https://www.openssl.org/policies/cla.html

**Comment by johnkohl-hcl** [comment]
> We will need a CLA for this. @johnkohl-hcl could you please submit a CLA - I assume we will need both ICLA and CCLA for you. https://www.openssl.org/policies/cla.html

Yes, I'm working through this with my employer (HCL Technologies).  We have an internal procedure that should have already resulted in a CCLA being submitted, and I'm going through the ICLA procedure now.

**Comment by johnkohl-hcl** [comment]
> We will need a CLA for this. @johnkohl-hcl could you please submit a CLA - I assume we will need both ICLA and CCLA for you. https://www.openssl.org/policies/cla.html

I've submitted my ICLA just now.  I'm still waiting on my company open-source coordinators to verify they've done the CCLA, and/or to have them submit it.

**Comment by johnkohl-hcl** [comment]
> > We will need a CLA for this. @johnkohl-hcl could you please submit a CLA - I assume we will need both ICLA and CCLA for you. https://www.openssl.org/policies/cla.html
> 
> I've submitted my ICLA just now. I'm still waiting on my company open-source coordinators to verify they've done the CCLA, and/or to have them submit it.

Sorry for the delay, I'm still pushing on my corporate team to get that submitted.

**Comment by johnkohl-hcl** [comment]
Still waiting for my corporate contacts to execute the CLA.  They say "soon".

**Comment by johnkohl-hcl** [comment]
> Corporate CLA has been submitted to [legal@opensslfoundation.org](mailto:legal@opensslfoundation.org)

Any word on whether legal team has reviewed the Corporate CLA and can remove the hold on this PR?

**Comment by johnkohl-hcl** [comment]
The "hold: cla required" tag on this PR can be removed now, right?

**Comment by t8m** [comment]
@johnkohl-hcl we've received the CCLA but we need also ICLA from you unless you've already sent it.

**Comment by johnkohl-hcl** [comment]
> @johnkohl-hcl we've received the CCLA but we need also ICLA from you unless you've already sent it.

See above, July 26th I submitted my ICLA

**Comment by t8m** [comment]
@mattcaswell can you please look where the ICLA got lost?

**Comment by johnkohl-hcl** [comment]
> @mattcaswell can you please look where the ICLA got lost?

Any news?

**Comment by johnkohl-hcl** [comment]
Anything more I can do to help here?  I could rebase to latest and force-push if you like?

**Comment by t8m** [comment]
As there are no conflicts, there is no need to rebase.

Ping @openssl/committers for second review.

**Review by t-j-h — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**
